build_tools/cmake/iree_bytecode_module.cmake

```diff
@@ -70,6 +70,20 @@ function(iree_bytecode_module)
 
   iree_get_executable_path(_COMPILE_TOOL_EXECUTABLE ${_COMPILE_TOOL})
 
+  if(CMAKE_SYSTEM_PROCESSOR STREQUAL "riscv" AND
+    RISCV_CPU STREQUAL "rv64" AND
+    NOT _RULE_FLAGS MATCHES "iree-llvm-target-triple")
+    # RV64 Linux crosscompile toolchain can support iree-compile with
+    # specific CPU flags. Add the llvm flags to support RV64 RVV codegen if
+    # llvm-target-triple is not specified.
+    list(APPEND _RULE_FLAGS "--iree-llvm-target-triple=riscv64")
+    list(APPEND _RULE_FLAGS "--iree-llvm-target-cpu=generic-rv64")
+    list(APPEND _RULE_FLAGS "--iree-llvm-target-abi=lp64d")
+    list(APPEND _RULE_FLAGS "--iree-llvm-target-cpu-features=+m,+a,+f,+d,+c,+v")
+    list(APPEND _RULE_FLAGS "--riscv-v-fixed-length-vector-lmul-max=8")
+    list(APPEND _RULE_FLAGS "--riscv-v-vector-bits-min=512")
```

Comment:
We have a bunch of these RISC-V flags in scripts and build files and I'd rather we centralize them instead of proliferating them further. Can we just make these the default in the compiler, or load them from the environment somehow?

Related questions:
* When would users be expected to change them?
* Are they currently specific to a particular device or emulator that we use in the project?

Reply:
The abi flag goes with the arch and os. In RISCV64, it has to be ld64d to support the hardware double library. The cpu feature flags are very flexible, depending on which ISA extension you would like to enable. For Linux it requires multiplication, atomic, single and double float, and compression ISA extensions. The vector extension goes with the last two flags to define the SIMD behavior.

Because RISCV is an open arch with so many different flavors, the intention here is to specify a set of flags for us to test the RVV codegen behavior, which @dcaballe spends most of the time developing/testing. The emulator actually enables all the ratified ISA extensions, and not just this set. The user, or different RISCV platforms, may have a different set of extensions, and it is up to them to select the appropriate set to enable the extensions properly.

However, you are absolutely right that we should avoid coding this length set over and over. I will consolidate that into a `IREE_RV64_DEFAULT_TEST_TARGET_CONFIGS` and put it in `riscv.toolchain.cmake`